source/extensions/matching/input_matchers/ip/matcher.cc

```diff
@@ -0,0 +1,37 @@
+#include "extensions/matching/input_matchers/ip/matcher.h"
+
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace InputMatchers {
+namespace IP {
+
+Matcher::Matcher(std::vector<Network::Address::CidrRange>&& ranges)
+    // We could put "false" instead of "true". What matters is that the IP
+    // belongs to the trie. We could further optimize the storage of LcTrie in
+    // this case by implementing an LcTrie<void> specialization that doesn't
+    // store any associated data.
+    : trie_({{true, std::move(ranges)}}) {}
+
+bool Matcher::match(absl::optional<absl::string_view> input) {
+  if (!input) {
+    return false;
+  }
+  const absl::string_view& ip_str = *input;
```

Comment:
I think we normally just copy string_views around since they are so small

Reply:
Done in https://github.com/envoyproxy/envoy/pull/16592/commits/04b5afe96f8838eabb1d4e38e64b590439df30ff